From fcf5506c93520306fb8d3cc52c964792c73c060f Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Fri, 08 Mar 2024 08:30:09 +0200
Subject: [PATCH] [SW-156330] gaudi3: pldm: increase ppboot size

When TPM code is added for gaudi3, the ppboot image
for pldm is found to not fit within the 16kb size
limit. Therefore, as preparation for TPM support
inclusion, increasing ppboot size limit for pldm
to 17kb.

Change-Id: I54c6a2df6a6a1d6bb5e2daec5610f2c8afe3ffa1
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index cbde106..5e3d172 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -1,7 +1,11 @@
 include $(HL_DEV_ARCH_COMMON)/Makefile
 include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
 
-PPBOOT_MAX_SIZE := 16384
+ifeq ($(HL_TARGET),pldm)
+	PPBOOT_MAX_SIZE := 17408
+else
+	PPBOOT_MAX_SIZE := 16384
+endif
 
 ifeq ($(HL_TARGET),onix)
 	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports_onix.elf
diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index cb144ba..066c324 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -8,7 +8,11 @@
 
 #include <sizes.h>
 
+#ifdef TARGET_IS_PLDM
+#define PPBOOT_SIZE			0x4400		/* 17KB */
+#else
 #define PPBOOT_SIZE			0x4000		/* 16KB */
+#endif
 #define PREBOOT_SIZE			0x20000		/* 128KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x60000
